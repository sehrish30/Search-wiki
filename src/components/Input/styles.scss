.input-field{
    padding: 15px 20px;
    border: 2px solid #bedbbb;
    border-radius: 30px;
    font-size: 24px;
    display: flex;
    z-index: 3;
    height: 3rem;
    padding: 0 1.5rem;
    width: 35rem;
    box-sizing: border-box;

    &:focus{
        box-shadow: 0 1px 6px 0 #707070;
        outline: none;
        border-color: #bedbbb
    }
}