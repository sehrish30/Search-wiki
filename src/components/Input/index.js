// npm install --save enzyme enzyme-adapter-react-16 react-test-renderer
// also added jest section in package.json
// npm i enzyme-to-json

// npm test -- --coverage --watchAll=false

import './styles.scss'
const Input = ({placeholder, ...rest}) => <input className="input-field" placeholder={placeholder} {...rest}/>

Input.defaultProps = {
    placeholder: 'Input search query',
};

export default Input;

