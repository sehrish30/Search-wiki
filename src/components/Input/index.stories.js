import React from 'react';
import Input from '.';


export default {
  title: 'Components/Input',
  component: Input,
//   argTypes: {
//     backgroundColor: { control: 'color' },
//   },
};

const Template = (args) => <Input {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  placeholder: 'Search data'
};

// export const Secondary = Template.bind({});
// Secondary.args = {
//   label: 'Button',
// };

// export const Large = Template.bind({});
// Large.args = {
//   size: 'large',
//   label: 'Button',
// };

// export const Small = Template.bind({});
// Small.args = {
//   size: 'small',
//   label: 'Button',
// };
