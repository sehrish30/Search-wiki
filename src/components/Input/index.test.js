import React from 'react'
import { shallow } from 'enzyme'

import Input from '.'

const render = (props) => shallow(<Input {...props} />)

describe('Input Component', ()=> {
    let sut;
    let props;

    describe('without props', () => {
         // Input components result props should match snapshot
        beforeEach(() => {
            props = {
                placeholder: "Input search query",
                name: 'search'
            };
            sut = render(props);
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        })
    });

   
})
