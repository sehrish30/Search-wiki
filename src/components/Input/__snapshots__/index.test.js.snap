// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component without props should match snapshot 1`] = `
<input
  className="input-field"
  name="search"
  placeholder="Input search query"
/>
`;
